_episodes/06-raster-intro.md

```diff
@@ -0,0 +1,462 @@
+
+An option to distinguish missing values from real data, is to use `nan` values to represent them. This can be done by specifying `masked=True` when loading the raster:
+~~~
+raster_ams_b9 = rioxarray.open_rasterio(items[0].assets["B09"].href, masked=True)
+~~~
+{: .language-python}
+
+One can also use the `where` function to select all the pixels which are different from the `nodata` value of the raster:
+~~~
+raster_ams_b9.where(raster_ams_b9!=raster_ams_b9.rio.nodata)
+~~~
+{: .language-python}
+
+Either way will change the `nodata` value from 0 to `nan`. Now if we do the statistics again, we will see the difference:
+~~~
+print(raster_ams_b9.min())
+print(raster_ams_b9.max())
+print(raster_ams_b9.mean())
+print(raster_ams_b9.std())
+~~~
+{: .language-python}
+~~~
+<xarray.DataArray ()>
+array(8., dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(15497., dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(2477.405, dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(2061.9539, dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+~~~
+{: .output}
+
+And if we plot the image, the `nodata` pixels will show as white, since they are not 0 anymore:
+
+![Raster plot with rioxarray using the robust setting no data](../fig/E06-03-overview-plot-B09-robust-with-nan.png)
+
+## Raster Bands
+So far we looked into a single band, `B09` of a Sentinel-2 scene. However, to get a better understanding of the AoI, one may also want to visualize a true-color overview of the region. Then the single band will not be sufficient in this case.
+
+![Multi-band raster image](../images/dc-spatial-raster/single_multi_raster.png)
+
+The `overview` asset in the Sentinel-2 scene is a multiband asset. Similar to `B09`, we can load it by its `href`:
+~~~
+raster_ams_overview = rioxarray.open_rasterio(items[0].assets['overview'].href)
+~~~
+{: .language-python}
+~~~
+<xarray.DataArray (band: 3, y: 343, x: 343)>
+[352947 values with dtype=uint8]
+Coordinates:
+  * band         (band) int64 1 2 3
+  * x            (x) float64 6.002e+05 6.005e+05 ... 7.093e+05 7.096e+05
+  * y            (y) float64 5.9e+06 5.9e+06 5.899e+06 ... 5.791e+06 5.79e+06
+    spatial_ref  int64 0
+Attributes:
+    _FillValue:    0.0
+    scale_factor:  1.0
+    add_offset:    0.0
+~~~
+{: .output}
+
+The band number comes first when GeoTiffs are read with the `.open_rasterio()` function. As we can see in the `xarray.DataArray` object, the shape is `(band: 3, y: 343, x: 343)` now. There are three bands in the `xarray.DataArray` now. One can also check the shape in the `.shape` attribute:
+~~~
+raster_ams_overview.shape
+~~~
+{: .language-python}
+~~~
+(3, 343, 343)
+~~~
+{: .output}
+
+It's always a good idea to examine the shape of the raster array you are working with and make sure it's what you expect. Many functions, especially ones that plot images, expect a raster array to have a particular shape.
```

Comment:
This sentence should move above. For example before the sentence `One can also check the shape in the .shape attribute`.

Reply:
Thx! I moved this entence